src/kOS/Screen/TermWindow.cs

```diff
@@ -256,6 +258,15 @@ private void GetFontIfChanged()
 
                 terminalLetterSkin.label.font = font;
                 terminalLetterSkin.label.fontSize = fontSize;
+                if (recalcWidth)
+                {
+                    CharacterInfo chInfo;
+                    terminalLetterSkin.label.font.RequestCharactersInTexture("X"); // Make sure the char in the font is lazy-loaded by Unity.
+                    terminalLetterSkin.label.font.GetCharacterInfo('X', out chInfo);
+                    shared.Screen.CharacterPixelWidth = chInfo.advance;
+
+                    NotifyOfScreenResize(shared.Screen);
+                }
```

Comment:
Moving the logic here also had another desirable effect:  It made the terminal now retain its previous width/height (in chars, not pixels) when the font changes.  Previously I had bent over backward to try to keep the pixel size the same and choose a new cell count (widthXheight) to hold the font, but that was stupid and never worked that well,  Now the terminal just draws bigger or smaller when you change the font, to keep the same number of rows and columns.

(I had previously worried about pushing the bottom of the window offscreen when you pick a bigger font, so you can't reach the resize corner, which was why I did that convoluted logic.  But in retrospect, you can drag the window from anywhere within the window, so this is not a problem - you can just drag the window up until you reach the resize corner.)